### PR TITLE
Serve static files like Pages does

### LINKED ISSUE #38: [pages] serve should return a 404 response when no static asset is found.
This obviously messes with frameworks which do a:

```js
const response = await env.ASSETS.fetch()
if (response.ok) return response

// else
return new Response(dynamic)
```

which is pretty common. We should return 404s if they are the asset does not exist.

### PR DESCRIPTION
**Pull request by GregBrimble**
Implements a from-scratch 'toy' implementation of the Pages asset-server.

Supports:
- headers
- redirects
- asset-server

Still to do:
- asset-server headers

The implementations of headers, redirects and asset-server are 'toy' implementations based on how we do it in production. They are not 100% faithful. We might consider properly open-sourcing pieces of this in the future.

Fixes #38 & #45.

Also more gracefully terminates the child proxy process (#83 & #94).

And also auto-opens a browser window. Set `BROWSER=none` to disable.

### CODE REVIEW

File: packages/wrangler/package.json [hunk 1]
```diff
@@ -37,10 +37,11 @@
   ],
   "dependencies": {
     "@cloudflare/pages-functions-compiler": "0.3.8",
+    "chokidar": "^3.5.2",
```
Comment (threepointone) [inline packages/wrangler/package.json:40]: move this to devDependencies, it gets bundled so doesn't need to be here

File: packages/wrangler/package.json [hunk 2]
```diff
@@ -37,10 +37,11 @@
...
     "esbuild": "0.14.1",
     "miniflare": "2.0.0-rc.3",
     "semiver": "^1.1.0",
-    "serve": "^13.0.2"
+    "undici": "^4.11.1"
```
Comment (threepointone) [inline packages/wrangler/package.json:44]: move this to devDependencies, it gets bundled so doesn't need to be here

**Review by threepointone — CHANGES_REQUESTED** [review]
please run `npm install` at the root to regenerate package-lock.json? I got a bunch of integrity hashes added when I ran it locally.

**Comment by GregBrimble** [comment]
Done!

**Review by threepointone — APPROVED** [review]
aight, let's land this baby!

**Comment by threepointone** [comment]
oh wait, could you add a changeset please?

**Review by threepointone — CHANGES_REQUESTED** [review]
changeset please!

**Review by threepointone — APPROVED** [review]

**Pull request merged**